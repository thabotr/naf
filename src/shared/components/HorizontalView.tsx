import React from 'react';
import {View, ViewProps} from 'react-native';

export default function HorizontalView(props: ViewProps) {
  const style: Object = props.style?.valueOf() ?? {};
  const newProps: ViewProps = {
    ...props,
    style: {...style, display: 'flex', flexDirection: 'row'},
  };
  return <View {...newProps}>{props.children}</View>;
}
