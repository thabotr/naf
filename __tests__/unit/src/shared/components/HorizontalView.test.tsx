import React from 'react';
import renderer from 'react-test-renderer';
import HorizontalView from '../../../../../src/shared/components/HorizontalView';

describe('HorizontalView', () => {
  it('correctly renders its child components horizontally', () => {
    const tree = renderer.create(<HorizontalView>{}</HorizontalView>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
