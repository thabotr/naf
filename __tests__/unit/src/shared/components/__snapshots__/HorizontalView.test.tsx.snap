// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HorizontalView correctly renders its child components horizontally 1`] = `
<View
  style={
    {
      "display": "flex",
      "flexDirection": "row",
    }
  }
/>
`;
